geode-core/src/test/java/org/apache/geode/distributed/LocatorLauncherLocalIntegrationTest.java

```diff
@@ -14,745 +14,270 @@
+        .hasCauseInstanceOf(BindException.class);
   }
 
   @Test
-  public void testStartUsingPortInUseFails() throws Throwable {
-    // Test makes no sense in this case
-    if (this.locatorPort == 0) {
-      return;
-    }
-
-    // generate one free port and then use it instead of default
-    this.socket =
-        SocketCreatorFactory.getSocketCreatorForComponent(SecurableCommunicationChannel.CLUSTER)
-            .createServerSocket(this.locatorPort, 50, null, -1);
-
-    this.launcher = new Builder().setMemberName(getUniqueName()).setPort(this.locatorPort)
-        .setRedirectOutput(true).setWorkingDirectory(this.workingDirectory)
-        .set(CLUSTER_CONFIGURATION_DIR, this.clusterConfigDirectory).set(LOG_LEVEL, "config")
-        .build();
-
-    RuntimeException expected = null;
-    try {
-      this.launcher.start();
-      fail("LocatorLauncher start should have thrown RuntimeException caused by BindException");
-    } catch (RuntimeException e) {
-      expected = e;
-      assertNotNull(expected.getMessage());
-      // BindException string varies by platform
-    } catch (Throwable e) {
-      this.errorCollector.addError(e);
-    }
-
-    try {
-      assertNotNull(expected);
-      final Throwable cause = expected.getCause();
-      assertNotNull(cause);
-      assertTrue(cause instanceof BindException);
-      // BindException string varies by platform
-    } catch (Throwable e) {
-      this.errorCollector.addError(e);
-    }
-
-    try {
-      this.pidFile = new File(this.temporaryFolder.getRoot(), ProcessType.LOCATOR.getPidFileName());
-      assertFalse("Pid file should not exist: " + this.pidFile, this.pidFile.exists());
-
-      // creation of log file seems to be random -- look into why sometime
-      final String logFileName = getUniqueName() + ".log";
-      assertFalse("Log file should not exist: " + logFileName,
-          new File(this.temporaryFolder.getRoot(), logFileName).exists());
-
-    } catch (Throwable e) {
-      this.errorCollector.addError(e);
-    }
-
-    // just in case the launcher started...
-    LocatorState status = null;
-    try {
-      status = this.launcher.stop();
-    } catch (Throwable t) {
-      // ignore
-    }
-
-    try {
-      waitForFileToDelete(this.pidFile);
-      assertEquals(getExpectedStopStatusForNotRunning(), status.getStatus());
-    } catch (Throwable e) {
-      this.errorCollector.addError(e);
-    }
+  public void statusWithPidReturnsOnlineWithDetails() throws Exception {
+    givenRunningLocator();
+
+    LocatorState locatorState = new Builder().setPid(localPid).build().status();
+
+    assertThat(locatorState.getStatus()).isEqualTo(ONLINE);
+    assertThat(locatorState.getClasspath()).isEqualTo(getClassPath());
+    assertThat(locatorState.getGemFireVersion()).isEqualTo(GemFireVersion.getGemFireVersion());
+    assertThat(locatorState.getHost()).isEqualTo(InetAddress.getLocalHost().getCanonicalHostName());
```

Comment:
It could fail in the Geode Nightly Build too. We'll have to keep an eye on this...